qlever.py

```diff
@@ -0,0 +1,488 @@
+                print(f"Note: psutil.net_connections() failed ({e}),"
+                      f" will not scan network connections for action"
+                      f" \"start\"")
+
+        # Check whether docker is installed and works (on MacOS 12, docker
+        # hangs when installed without GUI, hence the timeout).
+        try:
+            completed_process = subprocess.run(
+                    ["docker", "info"], timeout=0.5,
+                    stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+            if completed_process.returncode != 0:
+                raise Exception("docker info failed")
+            self.docker_enabled = True
+        except Exception:
+            self.docker_enabled = False
+            print("Note: `docker info` failed, therefore"
+                  " docker.USE_DOCKER=true not supported")
+
+        # Check if the QLever binaries work.
+        try:
+            subprocess.run([self.config['server']['binary'], "--help"],
+                           stdout=subprocess.DEVNULL,
+                           stderr=subprocess.DEVNULL)
+            subprocess.run([self.config['index']['binary'], "--help"],
+                           stdout=subprocess.DEVNULL,
+                           stderr=subprocess.DEVNULL)
+            self.binaries_work = True
+        except Exception:
+            print("Note: QLever binaries not found or failed, therefore"
+                  " docker.USE_DOCKER=false not supported")
+            self.binaries_work = False
+
+    def log_level_is_debug(self):
+        """ Return True if the log level is debug. """
+
+        return self.config['DEFAULT']['log-level'] == "debug"
+
+    def set_config(self, section, option, value):
+        """
+        Set a value in the config file (throws an exception if the
+        section or option does not exist).
+        """
+
+        if not self.config.has_section(section):
+            raise Exception(f"Section {section} does not exist in Qleverfile")
+        if not self.config.has_option(section, option):
+            raise Exception(f"Option {option} does not exist in section "
+                            f"{section} in Qleverfile")
+        self.config[section][option] = value
+
+    def get_total_file_size(self, paths):
+        """ Get the total size of all files in the given paths in GB. """
+
+        total_size = 0
+        for path in paths:
+            for file in glob.glob(path):
+                total_size += os.path.getsize(file)
+        return total_size / 1e9
+
+    def alive_check(self, port):
+        """ Check if a QLever server is running on the given port. """
+
+        message = "from the qlever script".replace(" ", "%20")
+        curl_cmd = f"curl -s http://localhost:{port}/ping?msg={message}"
+        exit_code = subprocess.call(curl_cmd, shell=True,
+                                    stdout=subprocess.DEVNULL,
+                                    stderr=subprocess.DEVNULL)
```

Comment:
Should we use a cross-platform python http library for these things and `subprocess` etc. only where really necessary?